<launch>

  <arg name="min_height" default="0.12" />
  <arg name="max_height" default="0.44" />
  <arg name="rot_p_gain" default="0.2" />
  <arg name="control_mode" default="False" />


  <!-- arudino for hand -->
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node" output="screen">
    <param name="port" value="/dev/arduino-uno" />
    <param name="baud" value="57600" />
  </node>

  <!-- arm -->
  <!-- kersling controller -->
  <node pkg="kresling" type="kresling_arm_controller.py" name="kersling_arm_controller" output="screen">
    <param name="min_height" value="$(arg min_height)" />
    <param name="max_height" value="$(arg max_height)" />

    <param name="rot_p_gain" value="$(arg rot_p_gain)" />
    <param name="min_roll" value="0.45" />
    <param name="min_pitch" value="0.45" />

    <param name="control_mode" value="$(arg control_mode)" />
  </node>
  <!-- <node pkg="kresling" type="dynamixel_interface.py" name="kersling_arm_controller" output="screen" /> -->

  <!-- dynamixel interface -->
  <include file="$(find kresling)/launch/dynamixel_controllers.launch" >
    <arg name="usb_port" value="/dev/dynamixel"/>
  </include >

  <!-- joy -->
  <node pkg="joy" type="joy_node" name="joy_node" output="screen" >
    <param name="dev" value="/dev/input/js0"/>
    <param name="coalesce_interval" value="0.05"/>
  </node>

  <!-- mocap -->
  <include file="$(find kresling)/launch/mocap.launch" />

</launch>
